src/components/calcite-radio-button/calcite-radio-button.tsx

```diff
@@ -0,0 +1,251 @@
+import {
+  Component,
+  Host,
+  h,
+  Listen,
+  Prop,
+  Element,
+  Watch,
+  Event,
+  EventEmitter,
+} from "@stencil/core";
+import { guid } from "../../utils/guid";
+
+@Component({
+  tag: "calcite-radio-button",
+  styleUrl: "calcite-radio-button.scss",
+  shadow: true,
+})
+export class CalciteRadioButton {
+  constructor() {
+    this.check = this.check.bind(this);
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties and Validators/Watchers
+  //
+  //--------------------------------------------------------------------------
+
+  /** The checked state of the radio button. */
+  @Prop({ mutable: true, reflect: true }) checked: boolean = false;
+  @Watch("checked")
+  onCheckedChange(newChecked: boolean, oldChecked: boolean) {
```

Comment:
Nitpick: These watcher methods read as event handlers. Not sure if that'd be misleading.